event/stream.go

```diff
@@ -0,0 +1,179 @@
+package event
+
+import (
+	"artk.dev/assume"
+	"artk.dev/asynctx"
+	"artk.dev/clone"
+	"context"
+	"sync"
+)
+
+var _ Observer[any] = (&Stream[any]{}).Observe
+
+// Stream is a thread-safe in-memory event stream.
+type Stream[Event any] struct {
+	consumerChannels  []chan<- eventMsg[Event] // 24 bytes on 64 bits.
```

Comment:
Can you explain this to me please?
[![dunno](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/WjlTemxb6oA4PgZFmj08/70e4facb-a079-46c8-bbe4-e7101e43de8c/dunno.gif)](https://app.graphite.dev//settings/meme-library?org=espert-dev)

Reply:
"64 bits" is a reference to the processor word size. "24 bytes" refers to the actual size in bytes of that field (3 words of 64 bits or 8 bytes each, one for the base pointer, another for the length, and a third for capacity).

I'm tracking these things because these functions use a fair amount of concurrency, and I'm trying to minimise the number of cache lines that get invalidated on the processor architectures that we are most likely to target. On most contemporary processors, cache lines are 64 bytes long.